From 699062a129473f7f9f4dd21dad30e04405d3ef24 Mon Sep 17 00:00:00 2001
From: Thomas Loimer <thomas.loimer@tuwien.ac.at>
Date: Sun, 1 Oct 2023 14:36:55 +0200
Subject: [PATCH 5/9] Fix compound objects preventing utf-8 text, ticket #158

When a figure is exported to postscript (or any format derived from
postscript), a preamble must be inserted into the postscript file if the
figure contains any text with code points beyond ascii. The function
scanning all text objects was reset by any compund object that did not
contain non-ascii text. Scan correctly and leave the flag indicating
non-ascii text raised when scanning compound objects.
---
 fig2dev/dev/genps.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fig2dev/dev/genps.c b/fig2dev/dev/genps.c
index 3cb0038..855aefb 100644
--- a/fig2dev/dev/genps.c
+++ b/fig2dev/dev/genps.c
@@ -2752,7 +2752,7 @@ note_text_needing_cmap(F_compound *obj)
 	}
 
 	for (o = obj->compounds; o != NULL; o = o->next)
-		needs_cmap = note_text_needing_cmap(o);
+		needs_cmap = note_text_needing_cmap(o) || needs_cmap;
 
 	return needs_cmap;
 }
-- 
2.34.1

